main.go

```diff
@@ -61,19 +68,148 @@ func wrapMain() error {
+	app.Version = fmt.Sprintf("1.0.0-%s", rev)
+	app.Flags = []cli.Flag{
+		cli.BoolFlag{
+			Name:   "dry-run",
+			Usage:  "do not apply the Kubernetes templates",
+			EnvVar: "PLUGIN_DRY_RUN",
+		},
+		cli.BoolFlag{
+			Name:   "verbose",
+			Usage:  "dry run disables docker push",
+			EnvVar: "PLUGIN_VERBOSE",
+		},
+		cli.StringFlag{
+			Name:   "token",
+			Usage:  "service account's JSON credentials",
+			EnvVar: "PLUGIN_TOKEN",
+		},
+		cli.StringFlag{
+			Name:   "gcloud-cmd",
+			Usage:  "alternative gcloud cmd",
+			EnvVar: "PLUGIN_GCLOUD_CMD",
+		},
+		cli.StringFlag{
+			Name:   "kubectl-cmd",
+			Usage:  "alternative kubectl cmd",
+			EnvVar: "PLUGIN_KUBECTL_CMD",
+		},
+		cli.StringFlag{
+			Name:   "project",
+			Usage:  "gcp project",
+			EnvVar: "PLUGIN_PROJECT",
+		},
+		cli.StringFlag{
+			Name:   "zone",
+			Usage:  "zone of the container cluster",
+			EnvVar: "PLUGIN_ZONE",
+		},
+		cli.StringFlag{
+			Name:   "cluster",
+			Usage:  "name of the container cluster",
+			EnvVar: "PLUGIN_CLUSTER",
+		},
+		cli.StringFlag{
+			Name:   "namespace",
+			Usage:  "Kubernetes namespace to operate in",
+			EnvVar: "PLUGIN_NAMEPSACE",
+		},
+		cli.StringFlag{
+			Name:   "template",
+			Usage:  "optional - template for e.g. deployments",
+			EnvVar: "PLUGIN_TEMPLATE",
+			Value:  ".kube.yml",
+		},
+		cli.StringFlag{
+			Name:   "secret-template",
+			Usage:  "optional - template for the secret object",
+			EnvVar: "PLUGIN_SECRET_TEMPLATE",
+			Value:  ".kube.sec.yml",
+		},
+		cli.StringFlag{
+			Name:   "vars",
+			Usage:  "variables to use in template",
+			EnvVar: "PLUGIN_VARS",
+		},
+		cli.StringFlag{
+			Name:   "secrets",
+			Usage:  "variables to use in secret_template. These are base64 encoded by the plugin.",
+			EnvVar: "PLUGIN_SECRETS",
+		},
+		cli.StringFlag{
+			Name:   "secrets-base64",
+			Usage:  "variables to use in secret_template. These should already be base64 encoded; the plugin will not do so.",
+			EnvVar: "PLUGIN_SECRETS_BASE64",
+		},
+		cli.StringFlag{
+			Name:   "drone-build-number",
+			Usage:  "variables to use in secret_template. These should already be base64 encoded; the plugin will not do so.",
+			EnvVar: "DRONE_BUILD_NUMBER",
+		},
+		cli.StringFlag{
+			Name:   "drone-commit",
+			Usage:  "variables to use in secret_template. These should already be base64 encoded; the plugin will not do so.",
+			EnvVar: "DRONE_COMMIT",
+		},
+		cli.StringFlag{
+			Name:   "drone-branch",
+			Usage:  "variables to use in secret_template. These should already be base64 encoded; the plugin will not do so.",
+			EnvVar: "DRONE_BRANCH",
+		},
+		cli.StringFlag{
+			Name:   "drone-tag",
+			Usage:  "variables to use in secret_template. These should already be base64 encoded; the plugin will not do so.",
+			EnvVar: "DRONE_TAG",
+		},
+	}
+
+	if err := app.Run(os.Args); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func run(c *cli.Context) error {
 	vargs := GKE{}
+	vargs.DryRun = c.Bool("token")
```

Comment:
This series of assignments is where the new inputs are translated to the old interface